From d5ba8eb47782854f0b6060401148790ee6c97eeb Mon Sep 17 00:00:00 2001
From: Steven Moreland <smoreland@google.com>
Date: Fri, 4 Dec 2020 21:13:03 +0000
Subject: [PATCH 5/5] [Q_asb_2021-03] libbinder: readString*Inplace SafetyNet
 (II)

SafetyNet logs (this time for failure case, instead of success case).

Bug: 172655291
Test: adb logcat -b events | grep snet # exactly one occurance w/ repro
(c/p'd from 34af0637666f43ae62040ad1bad76468423feba2)
Merged-In: I75ace071693c0a4579ed9477f7b9212a6e27c36d
Change-Id: I75ace071693c0a4579ed9477f7b9212a6e27c36d

Change-Id: I95ef79bac588ed19496b4d6c53e3ae1e4dcce085
---
 libs/binder/Parcel.cpp | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/libs/binder/Parcel.cpp b/libs/binder/Parcel.cpp
index d4f1052cd8..933eb78fab 100644
--- a/libs/binder/Parcel.cpp
+++ b/libs/binder/Parcel.cpp
@@ -2192,8 +2192,11 @@ const char16_t* Parcel::readString16Inplace(size_t* outLen) const
     if (size >= 0 && size < INT32_MAX) {
         *outLen = size;
         const char16_t* str = (const char16_t*)readInplace((size+1)*sizeof(char16_t));
-        if (str != nullptr && str[size] == u'\0') {
-            return str;
+        if (str != nullptr) {
+            if (str[size] == u'\0') {
+                return str;
+            }
+            android_errorWriteLog(0x534e4554, "172655291");
         }
     }
     *outLen = 0;
-- 
2.34.1

